src/agent/CMakeLists.txt

```diff
@@ -31,4 +31,5 @@ target_link_libraries(agent PRIVATE ${Boost_LIBRARIES})
 
 if(BUILD_TESTS)
     add_subdirectory(tests)
+    add_subdirectory(queue)
```

Comment:
delete later

Reply:
Could we already add it with the rest of the `add_subdirectory` outside of the `BUILD_TESTS` block? And even further, already add a member in the Agent class. I've found that there were some compilation problems trying to do this.

Reply:
done!